components/tidb_query/src/codec/mysql/time/datetime.rs

```diff
@@ -0,0 +1,1344 @@
+    }
+
+    /// Returns the day number
+    pub fn day(self) -> u32 {
+        self.get_day()
+    }
+}
+
+mod parser {
+    use super::*;
+    fn bytes_to_u32(input: &[u8]) -> Result<u32> {
+        let mut sum = 0u32;
+        for digit in input {
+            assert(digit.is_ascii_digit())?;
+            sum = sum
+                .checked_mul(10)
+                .and_then(|t| t.checked_add(u32::from(digit - b'0')))
+                .ok_or_else(Error::truncated)?;
+        }
+        Ok(sum)
+    }
+
+    fn digit1(input: &[u8]) -> Result<(&[u8], &[u8])> {
+        let end = input
+            .iter()
+            .position(|&c| !c.is_ascii_digit())
+            .unwrap_or_else(|| input.len());
+        assert(end != 0)?;
+        Ok((&input[end..], &input[..end]))
+    }
+
+    fn space1(input: &[u8]) -> Result<&[u8]> {
+        let end = input
+            .iter()
+            .position(|&c| !c.is_ascii_whitespace())
+            .unwrap_or_else(|| input.len());
+
+        assert(end < input.len())?;
+        Ok(&input[end..])
+    }
+
+    /// We assume that the `input` is trimed and is not empty.
+    fn split_components(input: &str) -> Result<Vec<&[u8]>> {
+        let mut buffer = input.as_bytes();
+
+        assert!(
+            !buffer.is_empty()
+                && !buffer.first().unwrap().is_ascii_whitespace()
+                && !buffer.last().unwrap().is_ascii_whitespace()
+        );
+
+        let mut components = vec![];
+
+        while !buffer.is_empty() {
+            let (mut rest, digits): (&[u8], &[u8]) = digit1(buffer)?;
+
+            components.push(digits);
+
+            if !rest.is_empty() {
+                // If a whitespace is acquired, we expect we have already collected ymd.
+                if rest[0].is_ascii_whitespace() {
+                    assert(components.len() == 3)?;
+                    rest = space1(rest)?;
+                }
+                // If a 'T' is acquired, we expect we have already collected ymd.
+                else if rest[0] == b'T' {
+                    assert(components.len() == 3)?;
+                    rest = &rest[1..];
+                }
+                // If a punctuation is acquired, move forward the pointer.
+                else if rest[0].is_ascii_punctuation() {
+                    rest = &rest[1..];
+                } else {
+                    assert(false)?;
+                }
+            }
+
+            buffer = rest;
+        }
+
+        if (components.len() == 7 || components.len() == 2)
+            && input.as_bytes()[input.len() - components.last().unwrap().len() - 1] != b'.'
+        {
+            assert(false)?;
+        }
+
+        Ok(components)
+    }
+
+    fn adjust_year(year: u32) -> u32 {
+        if year <= 69 {
+            2000 + year
+        } else if year >= 70 && year <= 99 {
+            1900 + year
+        } else {
+            year
+        }
+    }
+
+    /// return an array that stores [year, month, day, hour, minute, second]
```

Comment:
the `[year, month, day, hour, minute, second]` should wrap with `, or they will be parse as the frist part of link's markdown format

Reply:
Done